src/DomainNetMap.cpp

```diff
@@ -0,0 +1,344 @@
+                            GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        if (!nnet_json["trace_outputs"].is_array()) {
+            throw Exception("DomainNetMapImp::" + std::string(__func__) + 
+                            ": Neural net json must have a key \"trace_outputs\" "
+                            "whose value is an array.",
+                            GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        if (nnet_json["trace_outputs"].array_items().size()
+            != m_neural_net->get_output_dim()) {
+            throw Exception("DomainNetMapImp::" + std::string(__func__) + 
+                            ": Neural net output dimension must match the number of "
+                            "trace outputs.",
+                    GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        for (const auto &input : signal_it->second.array_items()) {
+            if (!input.is_string()) {
+                throw Exception("DomainNetMapImp::" + std::string(__func__) + 
+                                ": Neural net signal inputs must be strings.",
+                                GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+            }
+            m_signal_inputs.push_back({
+                    m_platform_io.push_signal(
+                            input.string_value(),
+                            domain_type,
+                            domain_index),
+                    NAN});
+        }
+
+        for (const auto &input : delta_it->second.array_items()) {
+            if (!input.is_array() || input.array_items().size() != 2
+                || !input[0].is_string() || !input[1].is_string()) {
+                throw Exception("DomainNetMapImp::" + std::string(__func__) + 
+                                ": Neural net delta inputs must be tuples of strings.",
+                                GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+            }
+            m_delta_inputs.push_back({
+                    m_platform_io.push_signal(
+                            input[0].string_value(),
+                            domain_type,
+                            domain_index),
+                    m_platform_io.push_signal(
+                            input[1].string_value(),
+                            domain_type,
+                            domain_index),
+                    NAN, NAN, NAN, NAN});
+        }
+
+        for (const auto &output : nnet_json["trace_outputs"].array_items()) {
+            if (!output.is_string()) {
+                throw Exception("DomainNetMapImp::" + std::string(__func__) + 
+                                ": Neural net trace outputs must be strings.",
+                                GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+            }
+            m_trace_outputs.push_back(output.string_value());
+        }
+    }
+
+    std::shared_ptr<DenseLayer> DomainNetMapImp::json_to_DenseLayer(const json11::Json &obj) const {
```

Comment:
```suggestion
    std::shared_ptr<DenseLayer> DomainNetMapImp::json_to_DenseLayer(const json11::Json &obj) const
    {
```
I marked all the spots that I saw, but please check for this in all the new code.  The existing style we have in the code base is as I suggested.  The position of this open curly is mixed in this PR.